apps/src/aichat/views/UserChatMessageEditor.tsx

```diff
@@ -1,20 +1,29 @@
-import React from 'react';
+import React, {useState, useContext} from 'react';
 import Button from '@cdo/apps/templates/Button';
 import moduleStyles from './userChatMessageEditor.module.scss';
+import {ChatWorkspaceContext} from './ChatWorkspace';
+import aichatI18n from '../locale';
 
 const UserChatMessageEditor: React.FunctionComponent = () => {
+  const [userMessage, setUserMessage] = useState<string>('');
+
+  const onSubmit = useContext(ChatWorkspaceContext)?.onSubmit;
+
   return (
     <div className={moduleStyles.UserChatMessageEditor}>
       <textarea
         className={moduleStyles.textArea}
-        placeholder="Add a chat message..."
+        placeholder={aichatI18n.userChatMessagePlaceholder()}
+        onChange={e => setUserMessage(e.target.value)}
+        value={userMessage}
       />
       <Button
         key="submit"
         text="Submit"
         icon="arrow-up"
         onClick={() => {
```

Comment:
I don't have a preference. @sanchitmalhotra126 - thoughts?

Reply:
Doesn't hurt! We should probably be using `useCallback` more in general, thanks Denys for keeping us on it :)

Reply:
Thanks to you both!

Reply:
Do one of you mind explaining why we should be using `useCallback` more?

Reply:
@fisher-alice sorry for long response.
Sure, `useCallback` memoizes callback functions, so instead of creating a new callback function on every component's render - we'll have one callback function across multiple renders(untill one of callback's dependencies'll change), meaning perfomance optimization.
Here're couple pages about `useCallback` if you're interested: https://www.w3schools.com/react/react_usecallback.asp
https://www.freecodecamp.org/news/better-react-performance-usecallback-vs-usememo/

Also, in my opinion, using useCallback makes code cleaner by separating views(jsx,props,etc) and logic parts of code(callbacks, parsing data/props, etc)

Reply:
Thank you for the explanation!